File: benchmarks/src/ruby/floyd_warshall.rb
n=150;INF=999999999;dist=Array.new(n){|i|Array.new(n){|j|i==j ?0:INF}}
seed=12345;def ri(m);$seed=($seed*16807)%2147483647;$seed%m;end;$seed=12345
n.times{|i|4.times{|j|t=((i*7+j*13+1)%n);dist[i][t]=1+((i+j)%50)if t!=i}}
start=Time.now.to_f*1000
n.times{|k|n.times{|i|dik=dist[i][k];if dik!=INF;n.times{|j|nd=dik+dist[k][j];dist[i][j]=nd if nd<dist[i][j]};end}}
reach=sm=0;n.times{|i|n.times{|j|if dist[i][j]<INF;reach+=1;sm+=dist[i][j];end}}
finish=Time.now.to_f*1000;puts"Result: #{reach}_#{sm%1000003}";puts"Time: #{(finish-start).round}ms"

File: benchmarks/src/nodejs/floyd_warshall.js
const n=150,INF=999999999;let dist=Array.from({length:n},(_,i)=>Array.from({length:n},(_,j)=>i==j?0:INF));
let seed=12345;function ri(m){seed=(seed*16807)%2147483647;return seed%m}
for(let i=0;i<n;i++)for(let j=0;j<4;j++){let t=(i*7+j*13+1)%n;if(t!=i)dist[i][t]=1+((i+j)%50)}
const start=Date.now();
for(let k=0;k<n;k++)for(let i=0;i<n;i++){let dik=dist[i][k];if(dik!=INF)for(let j=0;j<n;j++){let nd=dik+dist[k][j];if(nd<dist[i][j])dist[i][j]=nd}}
let reach=0,sum=0;for(let i=0;i<n;i++)for(let j=0;j<n;j++)if(dist[i][j]<INF){reach++;sum+=dist[i][j]}
const end=Date.now();console.log("Result: "+reach+"_"+(sum%1000003));console.log("Time: "+Math.floor(end-start)+"ms")
